click/core.py

```diff
@@ -700,6 +700,12 @@ def main(self, args=None, prog_name=None, complete_var=None,
             except (EOFError, KeyboardInterrupt):
                 echo(file=sys.stderr)
                 raise Abort()
+            except Exit as e:
```

Comment:
I would implement this so that the logic lives alongside "Abort". It's logically very similar.
This would mean it would be better if Exit didn't derive from ClickException